### PR TITLE
OSD-16588: Added Logic to Warn when Documentation does not matches Product type

### PR DESCRIPTION
**Pull request by tokalevasant**
This PR closes [OSD-16588](https://issues.redhat.com//browse/OSD-16588). This adds a check to servicelog to check whether documentation link sent with the SL is correct or not!

### CODE REVIEW

**Review by devppratik — CHANGES_REQUESTED** [review]

File: cmd/servicelog/post.go [hunk 1]
```diff
@@ -193,13 +195,30 @@ func (o *PostCmdOptions) Run() error {
...
+		// documentation link matches the cluster product (rosa, dedicated)
+		if !o.skipPrompts && docClusterType != "" {
+			clusterType := cluster.Product().ID()
+
+			if docClusterType != clusterType {
+				log.Info("The documentation link in the servicelog is for '", docClusterType, "' while the servicelog itself is for cluster type '", clusterType, "'.")
+				if !ocmutils.ConfirmPrompt() {
+					log.Info("Skipping cluster ID: ", cluster.ID(), ", Name: ", cluster.Name())
+					continue
+				}
```
[the remark this answers is not in the document]
Comment (tokalevasant) [reply cmd/servicelog/post.go:218]: Discussed with Pratik. No changes necessary

File: cmd/servicelog/post.go [hunk 2]
```diff
@@ -193,13 +195,30 @@ func (o *PostCmdOptions) Run() error {
...
 			continue
 		}
 
+		// if servicelog description contains a documentation link, verify that
+		// documentation link matches the cluster product (rosa, dedicated)
+		if !o.skipPrompts && docClusterType != "" {
+			clusterType := cluster.Product().ID()
+
+			if docClusterType != clusterType {
+				log.Info("The documentation link in the servicelog is for '", docClusterType, "' while the servicelog itself is for cluster type '", clusterType, "'.")
```
[the remark this answers is not in the document]
Comment (tokalevasant) [reply cmd/servicelog/post.go:214]: Discussed with Pratik. Changed the log level. No change necessary for the second comment.

[the remark this answers is not in the document]
Comment (tokalevasant) [reply cmd/servicelog/post.go]: Changed as suggested.

File: cmd/servicelog/post.go [hunk 3]
```diff
@@ -213,6 +232,28 @@ func (o *PostCmdOptions) Run() error {
...
+				// for proper comparison with cluster product types, return "osd"
+				// where "dedicated" is used in the documentation urls
+				p = "osd"
+			}
+			return p
+		}
+	}
+
+	return ""
+}
```
[the remark this answers is not in the document]
Comment (tokalevasant) [reply cmd/servicelog/post.go]: changed as suggested

**Pull request merged**